File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/BaseView.java
package com.remotearthsolutions.expensetracker.contracts;

public interface BaseView {
    void showAlert(String title,String message,String btnOk,String btnCancel,BaseView.Callback callback);
    void showToast(String message);
    boolean isDeviceOnline();

    void showProgress(String message);
    void hideProgress();

    interface Callback{
        void onOkBtnPressed();
        void onCancelBtnPressed();
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/BaseView.kt
package com.remotearthsolutions.expensetracker.contracts

interface BaseView {
    fun showAlert(
        title: String?,
        message: String?,
        btnOk: String?,
        btnCancel: String?,
        btnNeutral: String?,
        callback: Callback?
    )

    fun showToast(message: String?)
    val isDeviceOnline: Boolean
    fun showProgress(message: String?)
    fun hideProgress()
    interface Callback {
        fun onOkBtnPressed()
        fun onCancelBtnPressed() {
        }

        fun onNeutralBtnPressed() {
        }
    }
}